GCTC-NTGC/gc-digital-talent
# Bug/4300 pool advertisements already applied to

closes #4300 
adds validation to throw an error if the user has a non DRAFT application for a pool
adds a test too, as well as updates ApiMessages (definitely something to start doing more of)

testing:
ensure checks pass
- open `/browse/pools` as well as `/users/:id/applications`
- apply to a posting, should take you to the review application, assert this is reflected in My Applications
- go back to the poster and apply again, should change nothing from the last step
- complete and submit the application, ensure this is reflected in My Applications
- go back to the poster now and apply again 

### assert
- it goes nowhere
- toast appears with Intl message  "You have already submitted an application to this opportunity."
- application status unchanged in My Applications 

### **EDIT**
- now no toast appears, but rather the button should change and no longer be active
